src/script_opt/ZAM/Ops.in

```diff
@@ -1198,35 +1198,48 @@ eval	ConstructTableOrSetPre()
 
 direct-unary-op Record-Constructor ConstructRecord
 
-internal-op Construct-Record
+macro ConstructRecordPost()
+	auto& r = frame[z.v1].record_val;
+	Unref(r);
+	r = new RecordVal(cast_intrusive<RecordType>(z.t), init_vals);
```

Comment:
This seems obfuscated to me due to the `auto&` part, but this might just be me. 
```
	auto* r = frame[z.v1].record_val;
	Unref(r);
	frame[z.v1].record_val = new RecordVal(cast_intrusive<RecordType>(z.t), init_vals);
```

Reply:
FWIW, it's an idiom used fairly often in `Ops.in`, so I'll leave it.

Reply:
Ah, then sorry for the noise.